stellargraph/data/explorer.py

```diff
@@ -47,6 +48,62 @@ def _default_if_none(value, default, name, ensure_not_none=True):
     return value
 
 
+def _ensure_int(value, name, min_value):
+    if not isinstance(value, int) or value < min_value:
+        raise ValueError(
+            f"{name}: expected an integer greater than or equal to {min_value}, found: {value}"
+        )
+
+
+class RandomWalk(ABC):
+    """
+    Abstract base class for Random Walk classes. A Random Walk class must implement a ``run`` method
+    which takes an iterable of node IDs and returns a list of walks. Each walk is a list of node IDs
+    that contains the starting node as its first element.
+    """
+
+    def __init__(self, graph, seed=None):
+        if not isinstance(graph, StellarGraph):
+            raise TypeError("Graph must be a StellarGraph or StellarDiGraph.")
+
+        self.graph = graph
+        self._random_state, self._np_random_state = random_state(seed)
+
+    def _get_random_state(self, seed):
+        """
+        Args:
+            seed: The optional seed value for a given run.
+
+        Returns:
+            The random state as determined by the seed.
+        """
+        if seed is None:
+            # Restore the random state
```

Comment:
Is "restore" the right word here? Maybe

```suggestion
            # Use the class's random state
```